prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -599,9 +595,18 @@ enum ibv_wc_status efa_rdm_cq_process_wc_closing_ep(struct efa_ibv_cq *cq, struc
 	uint64_t wr_id = cq->ibv_cq_ex->wr_id;
 	enum ibv_wc_status status = cq->ibv_cq_ex->status;
 	enum ibv_wc_opcode opcode = efa_ibv_cq_wc_read_opcode(cq);
-	struct efa_rdm_pke *pkt_entry = (struct efa_rdm_pke *) wr_id;
+	struct efa_rdm_pke *pkt_entry;
 	int prov_errno;
 
+#if ENABLE_DEBUG
+	uint8_t gen = wr_id & (EFA_RDM_BUFPOOL_ALIGNMENT - 1);
+	wr_id &= ~(EFA_RDM_BUFPOOL_ALIGNMENT - 1);
```

Comment:
similar comment, can we have a static inline for this pkt_entry recovery and asserting